### DISCUSSION TITLE
Emscripten -- within Javascript VM

### DISCUSSION THREAD

**Discussion by tornikeo** [emscripten-core/emscripten#17098] [Q&A] [answered]
As the title says, **is it possible to compile the emscripten project itself as a JavaScript runnable wasm + js files**? 

I'm trying to build a serverless C++ compiler in JS. If I am able to run the `emcc` within JS environment, it would allow me to run any c++ program within the Chrome tab. 

Very inefficient, I know, but getting to that point would help my overall project a lot.

Much love to the dev team <3

**Comment by sbc100** [comment, marked as answer]
Some folks have tried in the past yes, with some success.  See https://github.com/jprendes/emception

**Reply by sbc100** [reply]
For more discussion on this see https://github.com/emscripten-core/emscripten/issues/6432
